lib/matplotlib/collections.py

```diff
@@ -608,18 +608,13 @@ def set_linestyle(self, ls):
             complete description.
         """
         try:
-            if isinstance(ls, str):
-                ls = cbook.ls_mapper.get(ls, ls)
-                dashes = [mlines._get_dash_pattern(ls)]
-            else:
-                try:
-                    dashes = [mlines._get_dash_pattern(ls)]
-                except ValueError:
-                    dashes = [mlines._get_dash_pattern(x) for x in ls]
-
-        except ValueError as err:
-            raise ValueError('Do not know how to convert {!r} to '
-                             'dashes'.format(ls)) from err
+            dashes = [mlines._get_dash_pattern(ls)]
+        except ValueError:
+            try:
+                dashes = [mlines._get_dash_pattern(x) for x in ls]
+            except ValueError as err:
+                raise ValueError('Do not know how to convert {!r} to '
```

Comment:
```suggestion
                raise ValueError(f'Do not know how to convert {ls!r} to dashes') from err
```

We should probably add a test here if we expect to hit this code-path. It looks like there is a lot of other error checking within `_get_dash_pattern`, so I think this is checking the fallback for an iterable of dash patterns in the collection?

Reply:
> I think this is checking the fallback for an iterable of dash patterns in the collection?

yes, I think so.  A side effect of my change is that it will now attempt to loop through a bad string, which it didn’t before.  So it fails slower.  I’m not sure how much that matters, but I think the code is simplest this way.